lib/data/interceptor/auth_interceptor.dart

```diff
@@ -0,0 +1,76 @@
+
+
+import 'package:dio/dio.dart';
+import 'package:flutter_template/data/model/auth/auth_status.dart';
+import 'package:flutter_template/data/model/auth/token.dart';
+import 'package:flutter_template/data/repository/auth_network_repository.dart';
+import 'package:flutter_template/data/repository/auth_storage_repository.dart';
+import 'package:flutter_template/extensions/extensions.dart';
+
+class AuthInterceptor extends InterceptorsWrapper {
+	final AuthNetworkRepository authNetworkRepository;
+	final AuthStorageRepository authStorageRepository;
+
+	AuthInterceptor(
+		this.authNetworkRepository,
+		this.authStorageRepository,
+		);
+
+	@override
+	Future onRequest(
+		RequestOptions options,
+		RequestInterceptorHandler handler)
+	async {
+		// Ignore auth related calls.
+		if (options.path.contains('auth') && !options.path.contains('/auth/password/change')) {
+			return handler.next(options);
+		}
+		// Ignore user registration calls.
+		if (!options.path.contains('users/device-tokens') &&
+			options.path.contains('users') &&
+			!options.path.contains('connect-stripe') &&
+			!options.path.contains('addresses') &&
+			options.method == 'POST') {
```

Comment:
Kinda project specific, maybe we should clean this out?

This one is from the Endeavour project, right? Think this is tailored to Casper's custom oauth solution, projects using standard oauth2 flows prolly want to use another interceptor. e.g. like the one we have on the Vim project -could be added in another PR.